sequencer/src/api.rs

```diff
@@ -1575,24 +1611,32 @@ mod test {
                 .send()
                 .await
                 .unwrap();
-
             for peer in &network.peers {
                 let state = peer.consensus().read().await.decided_state().await;
 
                 match state.chain_config.resolve() {
                     Some(cf) => {
-                        if cf != chain_config_upgrade && height as u64 > stop_voting_view {
-                            panic!("failed to upgrade chain config");
+                        // Fail test if we've waited long enough
+                        if height as u64 > stop_voting_view {
```

Comment:
this should be inside the if statement  `if cf != chain_config_upgrade`

Reply:
I think If we exceed the number of configured views, we should fail the test. If chain_config gets upgraded *after* that number, still a failure. No?

Reply:
I am not sure if this is an accurate way to determine that chain config upgraded after that block height. The api call may take some time ?

Reply:
On the other hand I'm thinking we should avoid writing such mind bending tests. I think part of the reason we didn't catch the bug in the test is that it was so hard to understand. So maybe we can think of going in a different direction.